profiles/profile.py

```diff
@@ -1,155 +1,156 @@
-    float_format=float_format,
-    compression=compression,
-)
-
-# Feature Selection (DMSO Control) - Level 4B Data
-feat_dmso_file = pathlib.PurePath(
-    output_dir, f"{plate_name}_normalized_feature_select_dmso.csv.gz"
-)
-feature_select(
-    profiles=norm_dmso_file,
-    features="infer",
-    operation=feature_select_ops,
-    output_file=feat_dmso_file,
-    float_format=float_format,
-    compression=compression,
-)
-
-# Feature Selection (Whole Plate) - Level 4B Data
-feat_file = pathlib.PurePath(
-    output_dir, f"{plate_name}_normalized_feature_select.csv.gz"
-)
-feature_select(
-    profiles=norm_file,
-    features="infer",
-    operation=feature_select_ops,
-    output_file=feat_file,
-    float_format=float_format,
-    compression=compression,
-)
\ No newline at end of file
+def process_profile(batch, plate, cell, pipeline):
+    # Set output directory information
+    pipeline_output = pipeline["output_dir"]
+    output_dir = pathlib.PurePath(".", pipeline_output, batch, plate)
+
+    # Set output file information
+    aggregate_output_file = pathlib.PurePath(output_dir, f"{plate}.csv.gz")
+    annotate_output_file = pathlib.PurePath(output_dir, f"{plate}_augmented.csv.gz")
+    normalize_output_file = pathlib.PurePath(output_dir, f"{plate}_normalized.csv.gz")
+    normalize_output_negcon_file = pathlib.PurePath(
+        output_dir, f"{plate}_normalized_negcon.csv.gz"
+    )
+    feature_output_file = pathlib.PurePath(
+        output_dir, f"{plate}_normalized_feature_select.csv.gz"
+    )
+    feature_output_negcon_file = pathlib.PurePath(
+        output_dir, f"{plate}_normalized_feature_select_negcon.csv.gz"
+    )
+
+    # Load pipeline options
+    compression = process_pipeline(pipeline["options"], option="compression")
+    float_format = process_pipeline(pipeline["options"], option="float_format")
+    samples = process_pipeline(pipeline["options"], option="samples")
+
+    # Load and setup platemap info
+    metadata_dir = pathlib.PurePath(".", "metadata", "platemaps", batch)
+    barcode_plate_map_file = pathlib.PurePath(metadata_dir, "barcode_platemap.csv")
+    barcode_plate_map_df = pd.read_csv(barcode_plate_map_file)
+    plate_map_name = barcode_plate_map_df.query(
+        "Assay_Plate_Barcode == @plate"
+    ).Plate_Map_Name.values[0]
+    plate_map_file = pathlib.PurePath(metadata_dir, "platemap", f"{plate_map_name}.txt")
+    plate_map_df = pd.read_csv(plate_map_file, sep="\t")
+    plate_map_df.columns = [
+        f"Metadata_{x}" if not x.startswith("Metadata_") else x
+        for x in plate_map_df.columns
+    ]
+    platemap_well_column = pipeline["platemap_well_column"]
+
+    # Annotate Profiles
+    annotate_steps = pipeline["annotate"]
+    annotate_well_column = annotate_steps["well_column"]
+    if annotate_steps["perform"]:
+        if annotate_steps["external"]:
+            external_df = pd.read_csv(
+                pathlib.PurePath(".", "metadata", "moa", annotate_steps["external"]),
```

Comment:
the `annotate_steps["external"]` must include a relative path without having to rely on `pathlib.PurePath(".", "metadata", "moa")`